src/plugins/orangehrmRecruitmentPlugin/Api/CandidateRejectAPI.php

```diff
@@ -0,0 +1,154 @@
+    public const PARAMETER_NOTE = 'note';
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     * @throws TransactionException
+     */
+    public function update(): EndpointResult
+    {
+        $this->beginTransaction();
+        try {
+            $candidateId = $this->getRequestParams()->getInt(
+                RequestParams::PARAM_TYPE_ATTRIBUTE,
+                self::PARAMETER_CANDIDATE_ID
+            );
+            $note = $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_NOTE
+            );
+
+            $candidateVacancy = $this->getCandidateService()
+                ->getCandidateDao()
+                ->getCandidateVacancyByCandidateId($candidateId);
+            $this->throwRecordNotFoundExceptionIfNotExist($candidateVacancy, CandidateVacancy::class);
+
+            $candidateVacancy->setStatus(
+                CandidateService::STATUS_MAP[WorkflowStateMachine::RECRUITMENT_APPLICATION_ACTION_REJECT]
+            );
+
+            $this->getCandidateService()->getCandidateDao()->saveCandidateVacancy($candidateVacancy);
+
+            $candidateHistory = new CandidateHistory();
+            $candidateHistory->getDecorator()->setCandidateById($candidateId);
+            $candidateHistory->getDecorator()->setVacancyById($candidateVacancy->getVacancy()->getId());
+            $candidateHistory->setCandidateVacancyName($candidateVacancy->getVacancy()->getName());
+            $candidateHistory->setAction(WorkflowStateMachine::RECRUITMENT_APPLICATION_ACTION_REJECT);
+            $candidateHistory->getDecorator()->setPerformedBy($this->getAuthUser()->getEmpNumber());
+            $candidateHistory->setPerformedDate($this->getDateTimeHelper()->getNow());
+            $candidateHistory->setNote($note);
+            $result = $this->getCandidateService()->getCandidateDao()->saveCandidateHistory($candidateHistory);
+
+            $this->commitTransaction();
+            return new EndpointResourceResult(CandidateHistoryDefaultModel::class, $result);
+        } catch (RecordNotFoundException $e) {
+            $this->rollBackTransaction();
+            throw $e;
+        } catch (Exception $e) {
+            $this->rollBackTransaction();
+            throw new TransactionException($e);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForUpdate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                self::PARAMETER_CANDIDATE_ID,
+                new Rule(Rules::ENTITY_ID_EXISTS, [Candidate::class]),
```

Comment:
```suggestion
                new Rule(Rules::POSITIVE),
```
I guess this is enough, since you are checking 
```php
            $candidateVacancy = $this->getCandidateService()
                ->getCandidateDao()
                ->getCandidateVacancyByCandidateId($candidateId);
            $this->throwRecordNotFoundExceptionIfNotExist($candidateVacancy, CandidateVacancy::class);
```
in `update`

Reply:
each candidate does not have assigned candidate vacancy. That is why I first checked whether candidate exists. If candidate exists, then we can check for candidate vacancy availability for that particular candidate and throw the exception if candidate vacancy is not exists